File: app/src/main/java/dnd/combattracker/adapters/viewholders/EncounterViewHolder.java
package dnd.combattracker.adapters.viewholders;


import android.database.Cursor;
import android.view.View;
import android.widget.TextView;

import dnd.combattracker.R;
import dnd.combattracker.repository.CombatTrackerContract;

public class EncounterViewHolder extends CustomViewHolder implements View.OnClickListener, View.OnLongClickListener {

    private TextView encounterNameTextView;

    public EncounterViewHolder(View view) {
        super(view);

        encounterNameTextView = (TextView) view.findViewById(R.id.list_item_encounter_name);

        //view.setLongClickable(true);
        view.setOnClickListener(this);
        view.setOnLongClickListener(this);
    }

    public void setData(Cursor c) {
        encounterNameTextView.setText(c.getString(c.getColumnIndex(CombatTrackerContract.EncounterEntry.NAME)));
    }

    @Override
    public void onClick(View view) {
        if (listener != null) {
            listener.onItemClick(view, getAdapterPosition());
        }
    }

    @Override
    public boolean onLongClick(View view) {
        if (longListener != null) {
            longListener.onLongItemClick(view, getAdapterPosition());
        }

        return true;
    }
}

File: app/src/main/java/tiru/initiative/adapters/EncounterViewHolder.kt
package tiru.initiative.adapters

import android.database.Cursor
import android.support.v7.widget.RecyclerView
import android.view.View
import kotlinx.android.synthetic.main.list_item_encounter.view.*
import tiru.initiative.repository.InitiativeTrackerContract

class EncounterViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
    fun setData(cursor: Cursor){
        itemView.list_item_encounter_name.text = cursor.getString(cursor.getColumnIndex(InitiativeTrackerContract.EncounterEntry.NAME))
    }
}